### DISCUSSION TITLE
v7: useWatch not working as expected

### DISCUSSION THREAD

**Discussion by surjithctly** [react-hook-form/react-hook-form#8503] [Q&A] [answered]
I forked the useWatch example from the official documentation, but it does not work as expected. See the GIF below. 

I have added `firstName` to watch, and it also updated the other field. But when I submitted the form, it fallback to default value. 

See it live:  https://codesandbox.io/s/react-hook-form-v7-usewatch-forked-jpph18?file=/src/index.js
See the GIF below. 

![CleanShot 2022-06-15 at 13 42 28](https://user-images.githubusercontent.com/1884712/173777733-977afcb6-59c5-4274-a605-daa3c18e9786.gif)

**Comment by bluebill1049** [comment, marked as answer]
can you share the link for the original codesandbox? will get it fixed.

**Reply by surjithctly** [reply]
This was the original link: https://codesandbox.io/s/react-hook-form-v7-usewatch-forked-9872t

**Reply by bluebill1049** [reply]
do this instead: https://codesandbox.io/s/react-hook-form-v7-usewatch-forked-3bso7f?file=/src/index.js

**Reply by surjithctly** [reply]
Thank you. It works. 

Any reason why the original method did not work? or am I being implemented it wrong? In earlier versions, it worked fine I think.

**Reply by bluebill1049** [reply]
when you provide `value` it will turn input into a controlled component, register is mostly/better applied to native and uncontrolled input.
